### PR TITLE
Use subtle-encoding for secure primitive Debug

### PR DESCRIPTION
**Pull request by maackle**

### CODE REVIEW

File: crates/holochain_zome_types/src/lib.rs [hunk 1]
```diff
@@ -175,21 +175,32 @@ macro_rules! secure_primitive {
 
         impl Eq for $t {}
 
+        #[cfg(not(feature = "subtle-encoding"))]
         /// The only meaningful debug information for a cryptograhpic secret is the literal bytes.
         /// Also, encodings like base64 are not constant time so debugging could open some weird
         /// side channel issue trying to be 'human friendly'.
         /// It seems better to never try to encode secrets.
         ///
+        /// Note that using this crate with feature "subtle-encoding", a hex
```
Comment (maackle) [inline crates/holochain_zome_types/src/lib.rs:184]:
```suggestion
        /// Note that when using this crate with feature "subtle-encoding", a hex
```

File: crates/holochain_zome_types/src/lib.rs [hunk 2]
```diff
@@ -175,21 +175,32 @@ macro_rules! secure_primitive {
...
+        /// representation will be used.
+        ///
         /// @todo maybe we want something like **HIDDEN** by default and putting the actual bytes
```
Comment (steveeJ) [inline crates/holochain_zome_types/src/lib.rs:187]: I think this PR takes care of the @todo here

**Review by steveeJ — APPROVED** [review]
:+1:

File: crates/holochain_zome_types/src/lib.rs (hunk 2 above)
Comment (thedavidmeister) [reply crates/holochain_zome_types/src/lib.rs:187]: @steveeJ i don't think so, subtle encoding just makes the encoding constant time, it doesn't hide any information

**Review by thedavidmeister — APPROVED** [review]

**Pull request merged**